### PR TITLE
orders coaches list

### PR DESCRIPTION
**Pull request by BriocheBerlin**
see #418

@PragTob mentioned that a test would be nice...

### CODE REVIEW

File: app/controllers/coaches_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class CoachesController < ApplicationController
   def index
-    @coaches = Person.where(workshop_coach: true)
+    # @coaches = Person.where(workshop_coach: true).order(:first_name, :last_name)
```
Comment (PragTob) [inline app/controllers/coaches_controller.rb]: #no_commented code

File: app/controllers/coaches_controller.rb [hunk 2]
```diff
@@ -9,3 +10,6 @@ def index
     @countries = @coaches.countries
   end
 end
+
+
+# user = User.ci_find('username', params[:username])
```
Comment (PragTob) [inline app/controllers/coaches_controller.rb]: #no_commented code

File: app/controllers/coaches_controller.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 class CoachesController < ApplicationController
   def index
-    @coaches = Person.where(workshop_coach: true)
+    # @coaches = Person.where(workshop_coach: true).order(:first_name, :last_name)
+    @coaches = Person.where(workshop_coach: true).order_by_name
```
Comment (PragTob) [inline app/controllers/coaches_controller.rb]: we could also implement `workshop_coach` as a scope :) (probably just coach or so)

**Comment by PragTob** [comment]
Yep would love to see a test that showcases basic sorting, the case insesitiveness and the sorting by first and last name :)

File: app/controllers/coaches_controller.rb (hunk 2 above)
Comment (berlintam) [inline app/controllers/coaches_controller.rb]: OOOH! what is ci_find??

File: spec/models/person_spec.rb [hunk 4]
```diff
@@ -143,6 +143,45 @@
...
+    let!(:workshop_coach) { create :workshop_coach }
+    let!(:second_workshop_coach)  { create :workshop_coach, first_name: 'Zlatan', last_name: 'Zukanovic'  }
+    let!(:third_workshop_coach)  { create :workshop_coach, first_name: 'lolo', last_name: 'lowercase' }
+    context 'combine scopes' do
+      it 'lists in alphabetical order when lowercased name is present' do
+        expect((subject.order_by_name).map(&:name)).to eq ['Corny Chocolate-Bar', 'lolo lowercase', 'Zlatan Zukanovic']
+      end
+    end
+  end
+
```
Comment (til) [inline spec/models/person_spec.rb]: Looks great! But maybe even a bit too much? Spec'ing the combination of two scopes is rather unusual I'd say. That the two scopes together work as expected when combined is the job of rails, and we should trust it that it gets it right.

File: spec/models/person_spec.rb [hunk 5]
```diff
@@ -143,6 +143,45 @@
...
+    let!(:second_person)  { create :second_person }
+
+    it 'lists the correct name' do
+      expect(subject).to contain_exactly workshop_coach
+    end
+
+    context 'for several workshop coaches' do
+      let!(:second_workshop_coach)  { create :workshop_coach, first_name: 'Zlatan', last_name: 'Zukanovic'  }
+      it 'lists in alphabetical order' do
+        expect(subject.map(&:name)).to eq ['Corny Chocolate-Bar', 'Zlatan Zukanovic']
```
Comment (til) [inline spec/models/person_spec.rb]: Are you sure that this is testing the right thing - isn't the order just random?

File: spec/models/person_spec.rb [hunk 6]
```diff
@@ -143,6 +143,45 @@
...
   end
 
+  describe '.workshop_coach' do
+    subject { described_class.workshop_coach }
+    let!(:workshop_coach) { create :workshop_coach }
+    let!(:second_person)  { create :second_person }
+
+    it 'lists the correct name' do
+      expect(subject).to contain_exactly workshop_coach
+    end
```
Comment (til) [inline spec/models/person_spec.rb]: Spec text and what it actually tests are a bit out of sync.

**Comment by BriocheBerlin** [comment]
Can we plühse merge this? 🐶 👀

File: .ruby-version [hunk 7]
```diff
@@ -1 +1 @@
-2.3.0
+2.1.5
```
Comment (til) [inline .ruby-version]: mep

**Comment by til** [comment]
Totally! Except the change to the `.ruby-version` file ...

**Comment by til** [comment]
Tadaaa!

**Pull request merged**